### ISSUE TITLE
Fix S107 FP: Don't raise on Zero-overhead member access methods

### ISSUE THREAD

**Issue by antonioaversa** [SonarSource/sonar-dotnet#8156]
A `static extern` method used as [C# 12 Zero-overhead member access](https://github.com/dotnet/runtime/issues/81741) method needs to match the signature of its target method or constructor.

If the target method/constructor has too many parameters (or exactly the maximum number of parameters), the zero-overhead member access method needs to match that or have one more (in the case of constructors). That results in S107 not being respected.

If the target method is not under the control of the owner of the zero-overhead member access method, there is nothing he can do to fix the issue, since signatures have to match.

For this reason, there should be an exception in presence of an `UnsafeAccessorAttribute`.

```cs
using System.Runtime.CompilerServices;

class ZeroOverheadMemberAccess
{
    [UnsafeAccessor(UnsafeAccessorKind.Constructor)]
    extern static UserData CallConstructor(int x1, int x2, int x3, int x4);                 // Noncompliant, FP: signature has to match target

    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "Method")]
    extern static void CallMethod(UserData userData, int x1, int x2, int x3, int x4);       // Noncompliant, FP: signature has to match target

    [UnsafeAccessor(UnsafeAccessorKind.StaticMethod, Name = "StaticMethod")]
    extern static void CallStaticMethod(UserData userData, int x1, int x2, int x3, int x4); // Noncompliant, FP: signature has to match target
}

class UserData
{
    UserData(int x1, int x2, int x3, int x4) { }         // Noncompliant

    void Method(int x1, int x2, int x3) { }              // Compliant
    static void StaticMethod(int x1, int x2, int x3) { } // Compliant
}
```

**Comment by antonioaversa** [comment]
Testing of this issue requires the UnitTest project to work with .NET 8, that is currently not possible due to issue described in this [Trello card](https://trello.com/b/1Zdg1PuD/net-8-c-12-spec-sonarway).
